File: 07BackendExamen/backend-examen-movil/api/models/Materia.js
/**
 * Materia.js
 *
 * @description :: A model definition represents a database table/collection.
 * @docs        :: https://sailsjs.com/docs/concepts/models-and-orm/models
 */

module.exports = {

  attributes: {

    nombre: {
      type: 'string',
      required: true,
      minLength: 3,
      maxLength: 60,
    },
    codigo: {
      type: 'string',
      required: true,
    },
    descripcion: {
      type: 'string',
      required: true,
    },
    activo: {
      type: 'boolean',
      required: true,
    },
    fechaCreacion: {
      type: 'string',
      required: true
    },
    numeroHorasPorSemana: {
      type: 'number',
      required: true
    },
    latitud: {
      type: 'string'
    },
    longitud: {
      type: 'string'
    },
    estudianteId: {
      model:'estudiante'
    }

  },

};

File: ExamenDef/app/src/main/java/com/example/examendef/Materia.kt
package com.example.examendef
import android.os.Parcel
import android.os.Parcelable
import java.util.*

class Materia (var id:Int,var nombre: String, var codigo:String, var descripcion:String,
               var activo:Boolean, var fechaCreacion:String,
               var numeroHorasPorSemana:Int, var estudianteId:Estudiante, var longitud: String, var latitud: String) {

    override fun toString(): String {
        return this.nombre
    }

}